crates/red_knot_python_semantic/resources/mdtest/type_properties/is_gradual_equivalent_to.md

```diff
@@ -46,6 +46,12 @@ static_assert(is_gradual_equivalent_to(Intersection[str | int, Not[type[Any]]],
 
 static_assert(not is_gradual_equivalent_to(str | int, int | str | bytes))
 static_assert(not is_gradual_equivalent_to(str | int | bytes, int | str | dict))
+
+# TODO: No errors
+# error: [static-assert-error]
+static_assert(is_gradual_equivalent_to(Unknown, Unknown | Any))
+# error: [static-assert-error]
+static_assert(is_gradual_equivalent_to(Unknown, Intersection[Unknown, Any]))
```

Comment:
I think the dynamic type reduction from your previous PR, which would fix these cases, is pretty solid (not overly special-cased) and could be put up as a separate PR?